extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -427,8 +428,10 @@ func (a *genericActuator) waitUntilWantedMachineDeploymentsAvailable(ctx context
 				return retryutils.Ok()
 			}
 
-			// scale down cluster autoscaler during creation or rolling update
-			if clusterAutoscalerUsed && !autoscalerIsScaledDown {
+			operationType = gardencorev1beta1helper.ComputeOperationType(worker.ObjectMeta, worker.Status.LastOperation)
+
+			// scale down cluster autoscaler during creation.
+			if clusterAutoscalerUsed && !autoscalerIsScaledDown && operationType == gardencorev1beta1.LastOperationTypeCreate {
```

Comment:
It is not necessary to scale down CA during creation because it is not even deployed yet (it will only be deployed after the machines came up). Hence, can we remove all the scale down code?

Reply:
Hm sure. I removed that part of the code which brings it down. I'll check the failure of `verify` step in ci.

Reply:
> Hm sure. I removed that part of the code which brings it down. I'll check the failure of verify step in ci.

You need to rebase to adopt https://github.com/gardener/gardener/pull/3347.

Reply:
I see, thank you @ialidzhikov, I just rebased it.